src/systems/user_commands/UserCommands.cc

```diff
@@ -1118,12 +1119,21 @@ bool CreateCommand::Execute()
           math::SphericalCoordinates::SPHERICAL,
           math::SphericalCoordinates::LOCAL2);
 
-      auto poseComp = this->iface->ecm->Component<components::Pose>(entity);
-      *poseComp = components::Pose({pos.X(), pos.Y(), pos.Z(), 0, 0,
-          IGN_DTOR(createMsg->spherical_coordinates().heading_deg())});
+      // Override pos and add to yaw
+      createPose.SetX(pos.X());
+      createPose.SetY(pos.Y());
+      createPose.SetZ(pos.Z());
+      createPose.Rot().Euler(
```

Comment:
I think there could potentially be an issue here if you are pitched completely up or down, the euler angles will reach a singularity and I don't think that yaw addition will work as expected.

Reply:
Ah good point, yay Euler angles. Honestly I'd like to remove the heading, but I put it there in the beginning, so I think it should keep working the same way if the rest of the rotation is zero. The main question is what is even expected in this situation...

Reply:
I'm using quaternion multiplication now, let me know what you think: 0fad8af908c51b33867f2d7986cfcf3b2d255afa

Reply:
* And documented on https://github.com/ignitionrobotics/ign-msgs/pull/212

Reply:
Yes, much safer, thank you!